src/MessageHandler/AbstractMessageHandler.php

```diff
@@ -24,10 +28,13 @@
      */
     protected $eventCallback;
 
-    public function __construct(Queue\Definition $queue, PredisClient $redisClient)
+    public function __construct(Queue\Definition $queue, PredisClient $redisClient, LoggerInterface $logger = null)
     {
         $this->queue = $queue;
         $this->redisClient = $redisClient;
+        if ($logger !== null) {
```

Comment:
Au lieu de faire des if à chaque fois, tu pourais faire un NullLogger, comme on le fait souvent.

Reply:
not too sure about this

if we want to be efficient, a simple `if` would better fit that a NullLogger

see https://github.com/M6Web/RedisMessageBroker/pull/9#pullrequestreview-75860941 for extra overhead coming with a NullLogger

if we don't want to be efficient, you're right, it's better to have a NullLogger

Reply:
I agree with you, but by adding a null logger your code is determinist and more simple to test, your code do always the same thing + in fact the overhead is insignificant. So, as you want, I replace my request changes by approuve.

Reply:
After some tests, having a `NullLogger` take x2 times than having an `if` (current solution).

Reply:
2 times of nothing is still nothing ;) :trollface:

Reply:
In our case no because this is Redis and we want to be faster 🏃 as we can, As we're going to push and take a looot of data.